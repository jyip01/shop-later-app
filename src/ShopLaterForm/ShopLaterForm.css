.ShopLater-form {
    background-color: darkgray;
    color: white;
    font-weight: bold;
    padding: 16px;
    max-width: 332px;
  }
  
  .ShopLater-form label {
    display: block;
    font-size: 20px;
    padding-bottom: 16px;
  }
  
  .ShopLater-form input,
  .ShopLater-form textarea,
  .ShopLater-form select {
    font-size: 14px;
    padding: 8px;
    width: 300px;
    border-radius: 5px;
  }
  
  .ShopLater-form select {
    height: 33px;
  }
  
  .ShopLater-form > .field {
    margin-bottom: 16px;
  }
  .ShopLater-form > .buttons {
    margin-top: 26px;
  }
  
  .ShopLater-form button {
    background-color: cornsilk;
    border-radius: 4px;
    border: 0;
    color: black;
    display: block;
    font-size: 18px;
    font-weight: bold;
    padding: 8px;
    text-align: center;
    text-decoration: none;
  }
  .ShopLater-form button[type="submit"]:hover {
    background-color: gray;
    color: white;
  }
  