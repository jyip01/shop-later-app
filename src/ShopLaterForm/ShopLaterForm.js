import React from 'react'
import './ShopLaterForm.css'

export default function ShopLaterForm(props) {
  const { className, ...otherProps } = props
  return (
    <form
      className={['ShopLater-form', className].join(' ')}
      action='#'
      {...otherProps}
    />
  )
}
