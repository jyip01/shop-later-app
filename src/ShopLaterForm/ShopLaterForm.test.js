import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ShopLaterForm from './ShopLaterForm'

describe(`ShopLaterForm component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  }

  it('renders a form.ShopLaterForm by default', () => {
    const wrapper = shallow(<ShopLaterForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the ShopLaterForm given props', () => {
    const wrapper = shallow(<ShopLaterForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
