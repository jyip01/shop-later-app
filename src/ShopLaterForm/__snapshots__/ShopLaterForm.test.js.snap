// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShopLaterForm component renders a form.ShopLaterForm by default 1`] = `
<form
  action="#"
  className="ShopLater-form "
/>
`;

exports[`ShopLaterForm component renders the ShopLaterForm given props 1`] = `
<form
  action="#"
  className="ShopLater-form test-class-name"
  data-other="test-other-prop"
>
  <p>
    test children
  </p>
</form>
`;
